axonserver/src/main/java/io/axoniq/axonserver/admin/eventprocessor/requestprocessor/LocalEventProcessorsAdminService.java

```diff
@@ -75,5 +76,56 @@ public LocalEventProcessorsAdminService(
                    .then();
         // the context will be removed from the event processor
     }
-}
 
+    @Nonnull
+    @Override
+    public Mono<Void> split(@Nonnull EventProcessorId identifier, @Nonnull Authentication authentication) {
+        String processor = identifier.name();
+        String tokenStoreIdentifier = identifier.tokenStoreIdentifier();
+        if (auditLog.isInfoEnabled()) {
+            auditLog.info("[{}] Request to split a segment of Event processor \"{}@{}\".",
+                          AuditLog.username(authentication.username()),
+                          processor,
+                          tokenStoreIdentifier);
+        }
+
+        EventProcessorIdentifier id = new EventProcessorIdentifier(processor, tokenStoreIdentifier);
+        return Flux.fromIterable(eventProcessors)
+                   .filter(eventProcessor -> id.equals(new EventProcessorIdentifier(eventProcessor)))
+                   .groupBy(ClientProcessor::context)
+                   .doOnNext(contextGroup -> contextGroup
+                           .map(ClientProcessor::clientId)
+                           .collectList()
+                           .subscribe(clients -> processorEventsSource.splitSegment(contextGroup.key(),
+                                                                                    clients,
+                                                                                    processor)))
+                   .then();
+        // the context will be removed from the event processor
+    }
+
+    @Nonnull
+    @Override
+    public Mono<Void> merge(@Nonnull EventProcessorId identifier, @Nonnull Authentication authentication) {
+        String processor = identifier.name();
+        String tokenStoreIdentifier = identifier.tokenStoreIdentifier();
+        if (auditLog.isInfoEnabled()) {
+            auditLog.info("[{}] Request to merge two segments of Event processor \"{}@{}\".",
+                          AuditLog.username(authentication.username()),
+                          processor,
+                          tokenStoreIdentifier);
+        }
+
+        EventProcessorIdentifier id = new EventProcessorIdentifier(processor, tokenStoreIdentifier);
+        return Flux.fromIterable(eventProcessors)
```

Comment:
whole method should probably be wrapped into Flux.defer() if `eventProcessors` list is mutable
Same for other methods

Reply:
why? the FluxIterable should simply keep a reference to the iterable and use it when someone subscribes.
Am I missing something?